lib/libutils/isoc/stack_check.c

```diff
@@ -3,13 +3,27 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 #include <compiler.h>
+#include <trace.h>
+
+#if defined(__KERNEL__)
+# include <kernel/panic.h>
+# define PANIC() panic("stack smashing detected")
```

Comment:
There already is a `stack smashing detected` error trace in `__stack_chk_fail()`.